MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,364 @@
+}
+```
+
+Now we can define queries directly through the method names (see the official documentation for the specific syntax).
+```java
+public interface PersonMongoRepository extends MongoRepository<Person, String> {
+  Optional<Person> findByName(String name);
+
+  List<Person> findAllByAgeIsLessThanEqual(int age);
+
+  List<Person> findAllByNameIsNot(String name);
+}
+```
+See the [official documentation](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.details) for further details.
+
+#### Query functions
+
+As Spring Data Mongo doesn't support/provide many query functions provided by Morphia, below are some replacements/alternatives.
+
+##### EqualIgnoreCase
+
+Morphia supports usage of _equalIgnoreCase()_. Use _regex()_ in Spring Data Mongo. For example
+
+Morphia - `query.criteria("fieldName").equalIgnoreCase(entity.getFieldName());`
+
+Spring Data Mongo - `query.addCriteria(where("fieldName").regex("^" + entity.getFieldName(), "i"));`
+
+##### HasAnyOf
+
+Morphia supports _hasAnyOf()_ method. Use _in()_ in Spring Data Mongo. For example 
+
+Morphia - `query.field("id").hasAnyOf(getIds());`
+
+Spring Data Mongo - `query.addCriteria(where("id").in(getIds()));`
+
+##### Filter
+
+Morphia supports _filter()_ method. Use _is()_ in Spring Data Mongo. For example
+
+Morphia - `query.filter("id", getId());`
+
+Spring Data Mongo - `query.addCriteria(where("id").is(getId()));`
+
+##### Contains
+
+Morphia supports _contains()_ method. Use _regex()_ in Spring Data Mongo. For example
+
+Morphia - `query.criteria("fieldName").contains(entity.getFieldName());`
+
+Spring Data Mongo - `query.addCriteria(where("fieldName").regex(".*" + entity.getFieldName() + ".*"));`
+
+##### HasAllOf
+
+Morphia supports _contains()_ method. Use _regex()_ in Spring Data Mongo. For example
+
+Morphia - `query.field("fieldName").hasAllOf(getFieldNameValues());`
+
+Spring Data Mongo - `query.addCriteria(where("fieldName").all(getFieldNameValues()));`
+
+
+### Error handling
+
+Morphia raised `ValidationException`s or subclasses of it when validation errors like duplicate
+unique keys occur at database level.
+Spring Data Mongo will throw more specific exceptions like
+`org.springframework.dao.DuplicateKeyException` or more generic
+`org.springframework.dao.DataAccessException`.
+Error handling must be updated.
+
+### Indexes
+
+While all `@Indexes` can be created at the main entity class in Morphia, each field must be
+`@Indexed`, `@TextIndexed` or `@WildcardIndexed` where it is defined in Spring Data Mongo.
+A `@CompoundIndex` can be declared at the entity class.
+
+### Custom converters
+
+Custom converters can be added directly to the Entity class in Morphia by using annotation like `@Converters(value = {ExampleConverter.class})`. 
+In Spring Data Mongo the custom converters can be added to the builder in your `Application.class`. The entity class that uses the converters must also be registered in the builder like below:
```

Comment:
I would remove mentioning to register the entity class. That is misleading because registering the entity and adding a converter is unrelated. Converters will work globally for all places where a specific type is found.

Reply:
updated